packages/warehouses/src/warehouseClients/TrinoWarehouseClient.ts

```diff
@@ -196,14 +196,21 @@ export class TrinoWarehouseClient extends WarehouseBaseClient<CreateTrinoCredent
                 );
             }
 
-            const result: QueryData = queryResult.value.data ?? [];
-
+            let result: QueryData = [];
             const schema: {
                 name: string;
                 type: string;
                 typeSignature: { rawType: string };
             }[] = queryResult.value.columns ?? [];
 
+            // Using `await` in this loop ensures data chunks are fetched and processed sequentially.
```

Comment:
Nice investigation! 🕵️ 
How many rows was necessary to have multiple chunks?

Reply:
I think it depends on how it's configured on trino side, this client calls the trino rest api to fetch the data.
https://trino.io/docs/current/develop/client-protocol.html
this endpoint returns the next uri address if needed.

Reply:
Thanks for the info. 🙌 
Can you check the failing tests? I suspect this is because the first chunk has `done: true` so it never enters the `while` loop and concats the results. 
We can also add a new unit test that simulates multiple chunks. You can mock multiple calls like this
```
jest.fn()
     .mockReturnValueOnce({ done: false, value: queryResponse })
     .mockReturnValueOnce({ done: true, value: queryResponse });
```

Reply:
sure, will do 👍